js/clipboard.js

```diff
@@ -28,20 +28,59 @@
 
```

Comment:
I agree with you that it seems unnecessary to have all those functions/constructors with the same name and how everything is created at the bottom. Without knowing a lot about this, I believe some of that is needed to be able to share the Clipboard with other JS files, like [clipboardToggleAll.js](https://github.com/artefactual/atom/blob/qa/2.x/js/clipboardToggleAll.js#L5), but I'll take a look.

Reply:
I'm not sure if passing the constructor that way has some benefits [for inheritance](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Inheritance#Setting_Teachers_prototype_and_constructor_reference), but in any case we don't need it. Specially the extra prototype and constructor created with `$.fn` at the bottom. All this is not needed for clipboardToggleAll.js either, where the association is made using the element data.

I've removed the unnecessary bits, but It would be great to be able to use [ES6 classes](https://hacks.mozilla.org/2015/07/es6-in-depth-classes/) at some point. I've created [this gist](https://gist.github.com/jraddaoui/85c7b8583af51158e6fabffa773f0890) using them (including current diff) and I'll mention it in SAD, but I guess [IE support](https://caniuse.com/#search=es6) is still a thing.

Reply:
Great thanks @jraddaoui. :)  I read the article you linked on ES6 classes and the changes look great.  It is too bad that ES6 classes aren't IE compatible - hopefully we can ditch IE support some time in the near future 🤞